packages/block-editor/src/autocompleters/block.js

```diff
@@ -1,85 +1,38 @@
 /**
  * External dependencies
  */
-import { once } from 'lodash';
+import { noop, map } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { select } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 import { createBlock } from '@wordpress/blocks';
+import { useMemo } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { searchBlockItems } from '../components/inserter/search-items';
+import useBlockTypesState from '../components/inserter/hooks/use-block-types-state';
+import { includeVariationsInInserterItems } from '../components/inserter/utils';
 import BlockIcon from '../components/block-icon';
 
-/** @typedef {import('@wordpress/block-editor').WPEditorInserterItem} WPEditorInserterItem */
-
-/** @typedef {import('@wordpress/components').WPCompleter} WPCompleter */
-
-/**
- * Returns the client ID of the parent where a newly inserted block would be
- * placed.
- *
- * @return {string} Client ID of the parent where a newly inserted block would
- *                  be placed.
- */
-function defaultGetBlockInsertionParentClientId() {
-	return select( 'core/block-editor' ).getBlockInsertionPoint().rootClientId;
-}
-
-/**
- * Returns the inserter items for the specified parent block.
- *
- * @param {string} rootClientId Client ID of the block for which to retrieve
- *                              inserter items.
- *
- * @return {Array<WPEditorInserterItem>} The inserter items for the specified
- *                                      parent.
- */
-function defaultGetInserterItems( rootClientId ) {
-	return select( 'core/block-editor' ).getInserterItems( rootClientId );
-}
-
-/**
- * Returns the name of the currently selected block.
- *
- * @return {string?} The name of the currently selected block or `null` if no
- *                   block is selected.
- */
-function defaultGetSelectedBlockName() {
-	const { getSelectedBlockClientId, getBlockName } = select(
-		'core/block-editor'
+const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
```

Comment:
It's copied in a few places, we should extract it and put in `@wordpress/blocks` as public API.

Reply:
Right

Reply:
Do you know if the "variations" are the only place where we could have this shape of innerBlocks.
It seems to me there's an inconsistency here. The variationIItem has a shape of a block object `{ name, innerBlocks, attributes }` but  its  innerBlocks have  a shape of a template `[ name, attributes, innerBlocks ]` which make me thing we should consolidate or rename `innerBlocks` to `innerBlocksTemplate` here.

Reply:
I just figured that it's because in the variations API we have innerBlocks property with the template shape too. I think that's not great as it's inconsistent with "example" API and it's inconsistent with it's self since a variation is an object with attributes and innerBlocks so you'd assume its innerBlocks keep the same shape.

I'll address this in a separate PR and I'll drop the template format from the variations while retaining backward compatibility.

Reply:
Yes, I agree that it is confusing. If you have some ideas on how to improve that, go for it :)

I thought about using so `innerBlocksTemplate` (I called the variable like that in the mapping helper) inside the variation item object but I left what was there before.